pkg/config/state_change.go

```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"errors"
+	"github.com/crc-org/vfkit/pkg/rest/define"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ErrNotImplemented Temporary Error Message
+var ErrNotImplemented = errors.New("function not implemented yet")
+
+// ChangeState execute a state change (i.e. running to stopped)
+func (vm *VirtualMachine) ChangeState(newState define.StateChange) error {
+	return ErrNotImplemented
+}
+
+// GetState returns state of the VM
+func (vm *VirtualMachine) GetState() VFVMState {
+	return VFVMState(vm.VzVM.State())
+}
+
+func (vm *VirtualMachine) Pause() error {
+	logrus.Debug("pausing virtual machine")
+	return vm.VzVM.Pause()
+}
+
+func (vm *VirtualMachine) Resume() error {
+	logrus.Debug("resuming machine")
+	return vm.VzVM.Resume()
+}
+
+func (vm *VirtualMachine) Stop() error {
+	logrus.Debug("stopping machine")
+	return vm.VzVM.Stop()
```

Comment:
`Stop()` is a  forceful stop of the VM. `RequestStop()` is the graceful stop.

Reply:
One caveat with the `Stop` API is that it's only supported with macOS 12+, an error will be returned on macOS 11. This can be wrapped in a `CanStop()` call, which will always return false on macOS 11. It's ok to fix that later in the future, I'm mentioning it while I have it in mind.

Reply:
fixed